### DISCUSSION TITLE
Pull request from dev branch to forked repo

### DISCUSSION THREAD

**Discussion by Rmh1978** [community/community#23088] [New to GitHub]
<p>Hi<br>
I am new to Github, but its amazing.</p>
<p>I am struggling it bit  with pull requests.</p>
<p>I have forked a repo with a open source webshop.<br>
In my own forked repo i have createt a branch for a new feature with 5 commits and have about 15 commits in my master branch.</p>
<p>I would now like to create a PR to the main repo for the webshop with only the commits from the feature branch with only the 5 commits, so it is easy for the owners of the main repo to check the PR/ commits for that feature only.</p>
<p>But when i try to create a PR from:<br>
Rmh1978 repo &gt; feature branch  to  Webshop repo &gt; master branch<br>
Then it takes all my commits, also the ones from my own master branch.<br>
If i try to PR:<br>
Rmh1978 repo &gt; feature branch  to  Rmh1978 repo &gt; master branch<br>
Then it works like i want it with only 5 commits.</p>
<p>What to do?<br>
Is my repo forked the wrong way or is my feature branch createt the wrong way.</p>
<p>The same a bit with PR on my master branch<br>
Rmh1978 repo &gt; master branch  to  Webshop repo &gt; master branch<br>
I have already made a PR, but when i then try to create another for new commits, it also include the commits that are allready made PR for.</p>

**Comment by Rmh1978** [comment, marked as answer]
<p>I found the error.</p>
<p>I had sourced the the feature branch from my own repo and not the main Webshop repo.<br>
Now it works like it want it.</p>

**Comment by tuves** [comment]
<p>Thank you for circling back to share your solutions, <a class="mention" href="/rmh1978">@Rmh1978</a>! It’ll help other users with the same issue find the answer quicker in the future :smile:</p>
